zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/appliers/TenantAppliersTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.appliers;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.engine.state.mutable.MutableMappingState;
+import io.camunda.zeebe.engine.state.mutable.MutableProcessingState;
+import io.camunda.zeebe.engine.state.mutable.MutableTenantState;
+import io.camunda.zeebe.engine.util.ProcessingStateExtension;
+import io.camunda.zeebe.protocol.impl.record.value.authorization.MappingRecord;
+import io.camunda.zeebe.protocol.impl.record.value.tenant.TenantRecord;
+import io.camunda.zeebe.protocol.record.value.EntityType;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(ProcessingStateExtension.class)
+public class TenantAppliersTest {
+
+  private MutableProcessingState processingState;
+
+  private MutableMappingState mappingState;
+  private MutableTenantState tenantState;
+  private TenantEntityAddedApplier tenantEntityAddedApplier;
+
+  @BeforeEach
+  public void setup() {
+    mappingState = processingState.getMappingState();
+    tenantState = processingState.getTenantState();
+    tenantEntityAddedApplier = new TenantEntityAddedApplier(processingState);
+  }
+
+  @Test
+  void shouldAddEntityToRoleWithTypeMapping() {
+    // given
+    final long entityKey = 1L;
+    mappingState.create(
+        new MappingRecord()
+            .setMappingKey(entityKey)
+            .setClaimName("claimName")
+            .setClaimValue("claimValue"));
+    final String tenantId = "tenantId";
+    final long tenantKey = 11L;
+    final var tenantRecord = new TenantRecord().setTenantId(tenantId).setTenantKey(tenantKey);
+    tenantState.createTenant(tenantRecord);
+    tenantRecord.setEntityKey(entityKey).setEntityType(EntityType.MAPPING);
+
+    // when
+    tenantEntityAddedApplier.applyState(tenantKey, tenantRecord);
+
+    // then
+    assertThat(tenantState.getEntityType(tenantKey, entityKey).get().equals(EntityType.MAPPING));
+    final var persistedMapping = mappingState.get(entityKey).get();
+    assertThat(persistedMapping.getTenantIdsList()).containsExactly(tenantId);
+  }
```

Comment:
❌ Could we add a same test case for users here? I know it's strictly not what this issue is about, but I suppose it was missed when user was implemented.

Reply:
Is the one above (?) 🤔

Reply:
There is only 1 test case in this class? What do you mean?

Reply:
I'll add a test case

Reply:
I thought I was going crazy 😄 Thank you!